src/nvim/decoration.c

```diff
@@ -840,14 +841,20 @@ void buf_signcols_count_range(buf_T *buf, int row1, int row2, int add, TriState
 
   // For each row increment "b_signcols.count" at the number of counted signs,
   // and decrement at the previous number of signs. These two operations are
-  // split in separate calls if "clear" is not kNone (surrounding a marktree splice).
+  // split in separate calls if "clear" is not kFalse (surrounding a marktree splice).
   for (int i = 0; i < row2 + 1 - row1; i++) {
-    int width = MIN(SIGN_SHOW_MAX, count[i] - add);
-    if (clear != kNone && width > 0) {
-      buf->b_signcols.count[width - 1]--;
-      assert(buf->b_signcols.count[width - 1] >= 0);
+    int prevwidth = MIN(SIGN_SHOW_MAX, count[i] - add);
+    if (clear != kNone && prevwidth > 0) {
+      if (buf->b_signcols.count[prevwidth - 1] > 0) {
+        buf->b_signcols.count[prevwidth - 1]--;
+      } else {  // Shouldn't happen, but not worth abort()ing over.
+        buf->b_signcols.count[prevwidth - 1] = 0;
+#ifndef NDEBUG
+        emsg("buf->b_signcols.count < 0");
+#endif
```

Comment:
Why is this not worth aborting? Even if it isn't, using `internal_error()` seems better.

Reply:
IDK maybe it is. I'm fine to keep the assert but aborting might make getting a way to reproduce easier.